note.go

```diff
@@ -0,0 +1,136 @@
+package hubspot
+
+const (
+	noteBasePath = "notes"
+)
+
+// NoteService is an interface of note endpoints of the HubSpot API.
+// HubSpot notes store information about interactions and communications.
+// It can be associated with other CRM objects such as contact, company, and deal.
+// Reference: https://developers.hubspot.com/docs/api/crm/notes
+type NoteService interface {
+	Get(noteID string, note interface{}, option *RequestQueryOption) (*ResponseResource, error)
+	Create(note interface{}) (*ResponseResource, error)
+	Update(noteID string, note interface{}) (*ResponseResource, error)
+	Delete(noteID string) error
+	AssociateAnotherObj(noteID string, conf *AssociationConfig) (*ResponseResource, error)
+}
+
+// NoteServiceOp handles communication with the note related methods of the HubSpot API.
+type NoteServiceOp struct {
+	notePath string
+	client   *Client
+}
+
+var _ NoteService = (*NoteServiceOp)(nil)
+
+// Note represents a note in HubSpot.
+type Note struct {
+	HsCreateDate   *HsTime `json:"hs_createdate,omitempty"`
+	HsObjectID     *HsStr  `json:"hs_object_id,omitempty"`
+	HsNoteBody     *HsStr  `json:"hs_note_body,omitempty"`
+	HsNoteTitle    *HsStr  `json:"hs_note_title,omitempty"`
+	HsTimestamp    *HsStr  `json:"hs_timestamp,omitempty"`
+	HubspotOwnerID *HsStr  `json:"hubspot_owner_id,omitempty"`
+}
+
+var defaultNoteFields = []string{
```

Comment:
Please let me know if you have any references for this as well.

Reply:
@kk-no HubSpot API documentation doesn't list them, but you an retrieve them from the properties API:

```bash
curl --request GET  --url 'https://api.hubapi.com/crm/v3/properties/note'  --header "authorization: Bearer $HUBSPOT_ACCESS_TOKEN"
```

Reply:
I understand. Thank you.
I remember having a similar problem before I released this library :(

Reply:
Please give me some time to check.